src/NationalInstruments.Analyzers/Style/LongChainDottedInvocationAnalyzer.cs

```diff
@@ -0,0 +1,115 @@
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecutionIf(IsRunningInProduction);
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(AnalyzeExpressionStatementSyntax, SyntaxKind.ExpressionStatement);
+            context.RegisterSyntaxNodeAction(AnalyzeEqualsValueClauseSyntax, SyntaxKind.EqualsValueClause);
+            context.RegisterSyntaxNodeAction(AnalyzeArgumentSyntax, SyntaxKind.Argument);
+            context.RegisterSyntaxNodeAction(AnalyzeArrowExpressionClauseSyntax, SyntaxKind.ArrowExpressionClause);
+        }
+
+        private static void AnalyzeExpressionStatementSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var expressionStatementSyntax = (ExpressionStatementSyntax)context.Node;
+
+            AnalyzeSyntax(expressionStatementSyntax, context);
+        }
+
+        private static void AnalyzeEqualsValueClauseSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var equalsValueClauseSyntax = (EqualsValueClauseSyntax)context.Node;
+
+            AnalyzeSyntax(equalsValueClauseSyntax, context);
+        }
+
+        private static void AnalyzeArrowExpressionClauseSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var expressionStatementSyntax = (ArrowExpressionClauseSyntax)context.Node;
+
+            AnalyzeSyntax(expressionStatementSyntax, context);
+        }
+
+        private static void AnalyzeArgumentSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var argumentSyntax = (ArgumentSyntax)context.Node;
+
+            AnalyzeSyntax(argumentSyntax, context);
+        }
+
+        private static void AnalyzeSyntax(SyntaxNode parentSyntaxNode, SyntaxNodeAnalysisContext context)
+        {
+            // Find the invocation expression i.e
+            // a method/delegate call or a property access
+            // or a chain of them
+            var invocationExpressionSyntax = parentSyntaxNode
+                .DescendantNodes()
+                .OfType<InvocationExpressionSyntax>()
+                .FirstOrDefault();
+
+            if (invocationExpressionSyntax is null)
+            {
+                // This does not contain any invocations, bail out
+                return;
+            }
+
+            // Find all invocations which involve brackets ()
+            var nonNestedInvocationsWithParenthesis = invocationExpressionSyntax
+                .DescendantNodes(syntaxNode => !syntaxNode.IsKind(SyntaxKind.Argument))
+                .OfType<ArgumentListSyntax>();
+
+            // There is a none or only a single call
+            if (nonNestedInvocationsWithParenthesis.Count() < 2)
+            {
+                // No more refactoring required
+                return;
+            }
+
+            // Find all end of lines in the expression
+            var endOfLineTrivias = invocationExpressionSyntax
```

Comment:
What happens if we have something like

```
objects.Where(x => {
   if (x > 2)
   {
      return true;
   }
   else
   {
       return false;
   }
}).Where(x => x % 2 == 0).Select(x => $"num {x}").First();
```

Reply:
Thanks. I missed this completely. I will update the logic in the analyzer and also add test to test this kind of code.